pkg/toolchains/git/git.bzl

```diff
@@ -0,0 +1,78 @@
+# Copyright 2021 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""toolchain to provide an git binary."""
+
+GitInfo = provider(
+    doc = """Information needed to invoke git.""",
+    fields = {
+        "name": "The name of the toolchain",
+        "valid": "Is this toolchain valid and usable?",
+        "label": "The path to a target I will build",
+        "path": "The path to a pre-built git",
+        "workspace_top": "The path to the top of the git workspace." +
+                         " In reality, use the path to the WORKSPACE file as a proxy for" +
+                         " a folder underneath the git workspace top.",
```

Comment:
How about 'git client'?

Reply:
I think of a "git client" as the "git" binary itself, not a "client workspace" like you would in perforce.

I think I would prefer "checkout" to either of these, but this could be changed later or supplemented with a comment if you like.

Reply:
There are not enough words!!!
I'm hopelessly biased on this one from having spent the last 13 years with perforce, where workspace and client are the overloaded words for "where your files are checked out".  A "checkout" command, adds them to your client.

I'm willing to look at this and change it later. It's not as if the UI will need a change. It's mostly the internal docs.